File: src/main/java/problems/TwoSum.java
package problems;

import java.util.HashMap;
import java.util.Map;

// https://leetcode.com/problems/two-sum/
public class TwoSum {

    public int[] twoSum(int[] nums, int target) {
        final Map<Integer, Integer> numTable = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (numTable.containsKey(target - nums[i])) {
                return new int[]{i, numTable.get(target - nums[i])};
            } else {
                numTable.put(nums[i], i);
            }
        }
        return new int[]{};
    }
}

File: python_problems/two_sum.py
"""
Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to
the target. You may assume that each input would have exactly one solution, and you may not use the same element twice.

https://leetcode.com/problems/two-sum/
"""


class TwoSum:
    def __init__(self, nums: [int], target: int):
        self.nums = nums
        self.target = target

    def two_sum(self) -> [int]:
        seen = {}
        for i, num in enumerate(self.nums):
            if self.target - num in seen:
                return [seen[self.target - num], i]
            seen[num] = i
